bears/general/IndentationBear.py

```diff
@@ -394,22 +399,23 @@ def get_valid_sequences(file,
 
             # ignore if within string
             for string in annotation_dict['strings']:
-                if(gt_eq(sequence_position, string.start) and
-                   lt_eq(sequence_position, string.end)):
+                if(gt_eq(sequence_position, string['full_range'].start) and
+                   lt_eq(sequence_position, string['full_range'].end)):
                     valid = False
 
             # ignore if within comments
             for comment in annotation_dict['comments']:
-                if(gt_eq(sequence_position, comment.start) and
-                   lt_eq(sequence_position, comment.end)):
+                if(gt_eq(sequence_position, comment['full_range'].start) and
+                   lt_eq(sequence_position, comment['full_range'].end)):
                     valid = False
 
-                if(comment.start.line == sequence_position.line and
-                        comment.end.line == sequence_position.line and
-                        check_ending):
-                    sequence_line_text = sequence_line_text[
-                        :comment.start.column - 1] + sequence_line_text[
-                        comment.end.column-1:]
+                if(comment['full_range'].start.line == sequence_position.line
```

Comment:
why would it not pass with dicts ?
`comment['full_range']` is sourcerange object.
and yes, ofcourse we are testing it.